configure.ac

```diff
@@ -220,13 +220,37 @@ AC_ARG_WITH([cpu],
 ],[
   case "$host_cpu" in
   powerpc*)
+    # Test successively newer processors and select the
+    # best implementation.
     AC_COMPILE_IFELSE([AC_LANG_SOURCE([[
       #ifdef _ARCH_PWR6
       # define SUCCESS 1
       #else
       # error "Support for hardware dfp isn't available"
       #endif
-      ]])], [with_dfp=yes], [with_dfp=no]) ;;
+      ]])], [with_dfp=yes; submachine=power6], [with_dfp=no])
+    AC_COMPILE_IFELSE([AC_LANG_SOURCE([[
+      #ifdef _ARCH_PWR6X
+      # define SUCCESS 1
+      #else
+      # error "Support for hardware dfp isn't available"
+      #endif
+      ]])], [with_dfp=yes; submachine=power6x], [])
+    AC_COMPILE_IFELSE([AC_LANG_SOURCE([[
+      #ifdef _ARCH_PWR7
```

Comment:
Do we really need to test for POWER7 and POWER8?
They both set _ARCH_PWR6 and _ARCH_PWR7.